cql3/statements/alter_table_statement.cc

```diff
@@ -404,13 +402,21 @@ std::pair<schema_builder, std::vector<view_ptr>> alter_table_statement::prepare_
                             cql3::dialect{});
                 }
             }
-            if (view_updated) {
-                builder.with_view_info(new_base_schema, view->view_info()->include_all_columns(), std::move(new_where));
-                view_updates.push_back(view_ptr(builder.build()));
-            }
+            // Even if no column is changed in the view, push an view update with the original where clause
+            // so that a new view schema is created with the new base schema.
+            builder.with_view_info(new_base_schema, view->view_info()->include_all_columns(), std::move(new_where));
+            view_updates.push_back(view_ptr(builder.build()));
```

Comment:
> statements: update all views when altering base schema 

I'm not sure this will work well in the context of a rolling upgrade. What if one of the old nodes updates the schema and a new node receives it? The code of the new version should take care to re-create the view schema objects in that case.

Moreover, the current approach in the PR is error prone - if somebody adds one more way to update the base table and they forget to update the views, you will have a bug.